drivers/virtualbox/vm.go

```diff
@@ -0,0 +1,55 @@
+package virtualbox
+
+import (
+	"bufio"
+	"io"
+	"strconv"
+	"strings"
+)
+
+type VirtualBoxVM struct {
+	CPUs   int
+	Memory int
+}
+
+func parseVMInfo(r io.Reader) (*VirtualBoxVM, error) {
+	s := bufio.NewScanner(r)
+	vm := &VirtualBoxVM{}
+	for s.Scan() {
+		line := s.Text()
```

Comment:
Yes, but then we would have to pass the ref to `VirtualBoxVm{}` to set the fields or make it generic to return the val and then parse.  I think it's concise the way it is and it's tested as well.